beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -562,6 +564,116 @@ impl Serialize for PeerConnectionStatus {
     }
 }
 
+/// Deserialization for http requests.
+impl<'de> Deserialize<'de> for PeerConnectionStatus {
```

Comment:
I think it would be great if we could implement `Serialize`/`Deserialize` without writing out our own visitor, which is prone to breakage if we add fields, and duplicates lots of the work done by `serde_derive`.

I think we could use `serde(with)` on the `Instant` fields, and `serde(tag = "status")` to set up an [internally-tagged enum representation](https://serde.rs/enum-representations.html#internally-tagged). We'll also likely need `#[serde(rename_all = "lowercase")` to lowercase the enum variant names.

We may want to break backwards compatibility while doing this, as that will allow us to drop the `last_seen: 0` shim and the `banned_ips` that's hardcoded to an empty `Vec`. AFAIK there's very little downstream code that depends on the current format, so we could change it slightly if it cleans up the code substantially.

Reply:
@AgeManning confirmed that we're OK to change the format slightly

Reply:
Thanks for your review @michaelsproul, I'm updating the PR.